scipy/signal/windows/windows.py

```diff
@@ -958,6 +927,96 @@ def barthann(M, sym=True):
 
     return _truncate(w, needs_trunc)
 
+def general_hamming(M, alpha, sym=True):
+    r"""Return a generalized Hamming window.
+
+    The generalized Hamming window is constucted by multiplying a rectangular
+    window by one period of a cosine function [1]_.
+
+    Parameters
+    ----------
+    M : int
+        Number of points in the output window. If zero or less, an empty
+        array is returned.
+    alpha : float
+        The window coefficient, :math:`\alpha`
+    sym : bool, optional
+        When True (default), generates a symmetric window, for use in filter
+        design.
+        When False, generates a periodic window, for use in spectral analysis.
+
+    Returns
+    -------
+    w : ndarray
+        The window, with the maximum value normalized to 1 (though the value 1
+        does not appear if `M` is even and `sym` is True).
+
+    Notes
+    -----
+    The generalized Hamming window is defined as
+
+    .. math:: w(n) = \alpha - \left(1 - \alpha\right) \cos\left(\frac{2\pi{n}}{M-1}\right)
+              \qquad 0 \leq n \leq M-1
+
+    Both the common Hamming window and Hann window are special cases of the
+    generalized Hamming window with :math:`\alpha` = 0.54 and :math:`\alpha` =
+    0.5, respectively [2]_.
+
+    See Also
+    --------
+    hamming, hann
+
+    Examples
+    --------
+
+    The Sentinel-1A/B Instrument Processing Facility uses generalized Hamming
+    windows in the processing of spaceborne Synthetic Aperture Radar (SAR)
+    data [3]_. The facility uses various values for the :math:`\alpha` parameter
+    based on operating mode of the SAR instrument. Some common :math:`\alpha`
+    values include 0.75, 0.7 and 0.52 [4]_. As an example, we plot these different
+    windows.
+
+    >>> from scipy.signal.windows import general_hamming
+    >>> from scipy.fftpack import fft, fftshift
+    >>> import matplotlib.pyplot as plt
+
+    >>> plt.figure()
+    >>> plt.title("Generalized Hamming Windows")
+    >>> plt.legend()
```

Comment:
[The legend is blank](https://5932-1460385-gh.circle-artifacts.com/0/html-scipyorg/generated/scipy.signal.windows.general_hamming.html), you need to call this after plotting the labeled curves